### PR TITLE
switch to 6.29.tikv branch

### PR DESCRIPTION
**Pull request by tabokie**
Titan patch: tikv/titan#246

Signed-off-by: tabokie <xy.tao@outlook.com>

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 	path = librocksdb_sys/rocksdb
 	url = https://github.com/tikv/rocksdb.git
-	branch = 6.4.tikv
+	branch = 6.29.tikv
 
 [submodule "titan"]
 	path = librocksdb_sys/libtitan_sys/titan
-	url = https://github.com/tikv/titan.git
-	branch = master
+	url = https://github.com/tabokie/titan.git
```
Comment (Connor1996) [inline .gitmodules:8]: update it

**Review by Connor1996 — APPROVED** [review]
don't forget to update the dependency later, LGTM

**Review by Connor1996 — COMMENTED** [review]
And is it being well tested?

**Comment by tabokie** [comment]
@Connor1996 Only the basics. QA won't run full tests since it's not scheduled on next release. My plan is to enable it on master first, then cut it back to 6.4 for releases.

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -2029,10 +2044,11 @@ void crocksdb_options_set_wal_bytes_per_sync(crocksdb_options_t* opt,
 static BlockBasedTableOptions* get_block_based_table_options(
     crocksdb_options_t* opt) {
   if (opt && opt->rep.table_factory != nullptr) {
-    void* table_opt = opt->rep.table_factory->GetOptions();
+    BlockBasedTableOptions* table_opt =
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/c.cc:2047]: Is it safe to assume it is `BlockBasedTableOptions`?

Comment (tabokie) [reply librocksdb_sys/crocksdb/c.cc:2047]: `GetOptions` will handle the dynamic typing.

Comment (BusyJay) [reply librocksdb_sys/crocksdb/c.cc:2047]: How? As far as I know, C++ template works at compile time.

Comment (tabokie) [reply librocksdb_sys/crocksdb/c.cc:2047]: Each type has a unique string id, each string id globally maps to an instance that can be hot loaded.

**Pull request merged**